atc/postgresrunner/join_limit_conn.go

```diff
@@ -0,0 +1,96 @@
+package postgresrunner
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"strings"
+
+	"github.com/Masterminds/squirrel"
+	"github.com/concourse/concourse/atc/db"
+	"github.com/onsi/ginkgo"
+)
+
+func validateJoinLimit(query string) error {
+	// Each subquery, split by a UNION or a UNION ALL, can have up to 8 JOINs
```

Comment:
Should we split out CTEs as well? I'm assuming since they're executed independently of the main query, they shouldn't contribute to the limit
```
WITH foo AS (
SELECT ... JOIN ... JOIN ... JOIN ... JOIN ...JOIN
),
SELECT ... JOIN ... JOIN ... JOIN ... JOIN
```

Reply:
Added cd1c046f70b5f741db93e75bbd60b8acc4f48bef that addresses CTEs and any subqueries (I don't *think* subqueries count toward the `JOIN` limit...). Happy to revert it as it's probably too complex for the benefit it adds - I was mostly just having fun with it.

Honestly, keeping the naïve `UNION` only splitting maybe isn't such a big deal, since we're at least guaranteed there's no false negatives.